.github/workflows/fuzz_test.yml

```diff
@@ -0,0 +1,132 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+name: Fuzz Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "core/src/**"
+      - "!core/src/docs/**"
+      - "!core/src/services/**"
+      - "core/src/services/fs/**"
+      - "core/src/services/memory/**"
+      - "core/src/services/s3/**"
+      - ".github/workflows/fuzz_test.yml"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+
+jobs:
+  fuzz-test-build-target:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/setup
+      - name: Install cargo fuzz
+        shell: bash
+        run: rustup install nightly && cargo +nightly install cargo-fuzz
+      - name: Build Fuzz Targets
+        shell: bash
+        working-directory: core/fuzz
+        run: cargo +nightly fuzz build -D
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: fuzz_targets
+          path: ./target/x86_64-unknown-linux-gnu/debug/fuzz_*
+  fuzz-test-run-s3:
+    runs-on: ubuntu-latest
+    needs: fuzz-test-build-target
+    services:
+      minio:
+        image: wktk/minio-server
+        ports:
+          - 9000:9000
+        env:
+          MINIO_ACCESS_KEY: "minioadmin"
+          MINIO_SECRET_KEY: "minioadmin"
+    strategy:
+      fail-fast: true
+      matrix:
+        fuzz-targets: [ fuzz_reader, fuzz_range_reader, fuzz_writer ]
+    steps:
+      - name: Setup Test Bucket
+        env:
+          AWS_ACCESS_KEY_ID: "minioadmin"
+          AWS_SECRET_ACCESS_KEY: "minioadmin"
+          AWS_EC2_METADATA_DISABLED: "true"
+        run: aws --endpoint-url http://127.0.0.1:9000/ s3 mb s3://test
+      - name: Download Fuzz Targets
+        uses: actions/download-artifact@v2
+        with:
+          name: fuzz_targets
+          path: ./target
+      - name: Run Fuzz Test
+        shell: bash
+        run: chmod 777 ./target/${{ matrix.fuzz-targets }} && ./target/${{ matrix.fuzz-targets }} -max_total_time=120
```

Comment:
Please use `chmod +x` instead of `777`. And I prefer to split them into different steps for easy undetstanding.